pkg/controllers/sidecar/identity/member.go

```diff
@@ -63,27 +66,43 @@ func Retrieve(ctx context.Context, name, namespace string, kubeclient kubernetes
 	}, nil
 }
 
-func (m *Member) GetSeeds(ctx context.Context, kubeClient kubernetes.Interface) ([]string, error) {
-	var services *corev1.ServiceList
-	var err error
+// GetSeed returns a single node contact point used for bootstrapping.
+// A single node is enough because bootstrap process will fail if any of the nodes in cluster is down.
+func (m *Member) GetSeed(ctx context.Context, coreClient v1.CoreV1Interface) (string, error) {
+	pods, err := coreClient.Pods(m.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			naming.ClusterNameLabel: m.Cluster,
+		}).String(),
+	})
+	if err != nil {
+		return "", err
+	}
 
-	sel := fmt.Sprintf("%s,%s=%s", naming.SeedLabel, naming.ClusterNameLabel, m.Cluster)
+	// Return static IP of itself to bootstrap first node.
+	if len(pods.Items) == 1 && pods.Items[0].Name == m.Name {
+		return m.StaticIP, nil
+	}
 
-	const maxRetryCount = 5
-	for retryCount := 0; ; retryCount++ {
-		services, err = kubeClient.CoreV1().Services(m.Namespace).List(ctx, metav1.ListOptions{LabelSelector: sel})
-		if err == nil && len(services.Items) > 0 {
-			break
-		}
-		if retryCount > 5 {
-			return nil, errors.New(fmt.Sprintf("failed to get seeds, error: %+v, len(services): %d", err, len(services.Items)))
+	// If at least one pod is ready, return the IP of service.
+	for _, pod := range pods.Items {
+		if podutils.IsPodReady(&pod) && pod.Name != m.Name {
```

Comment:
Pod can go unready just after this line so you can still potentially give a service to unready. Would be better to give scylla a bigger list of seeds and let it fallback to another one if it can't connect?